src/sentry/migrations/0459_control_files.py

```diff
@@ -0,0 +1,146 @@
+# Generated by Django 2.2.28 on 2023-05-24 05:27
+
+import django.db.models.deletion
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+import sentry.db.models.fields.jsonfield
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0458_add_options_source_column"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="ControlFile",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("name", models.TextField()),
+                ("type", models.CharField(max_length=64)),
+                (
+                    "timestamp",
+                    models.DateTimeField(db_index=True, default=django.utils.timezone.now),
+                ),
+                ("headers", sentry.db.models.fields.jsonfield.JSONField(default=dict)),
+                ("size", sentry.db.models.fields.bounded.BoundedPositiveIntegerField(null=True)),
+                ("checksum", models.CharField(db_index=True, max_length=40, null=True)),
+                ("path", models.TextField(null=True)),
+            ],
+            options={
+                "db_table": "sentry_controlfile",
+            },
+        ),
+        migrations.CreateModel(
+            name="ControlFileBlob",
```

Comment:
I don't know if we're going to need FileBlob, and FileBlobIndex for control silo files. The region implementation of files uses these models to segment files up into 1MB chunks. Our current use cases for files in control silo are all small files (avatars, policy documents) and these additionals models are additional complexity that we aren't going to use.